### ISSUE TITLE
Opus codec ?

### ISSUE THREAD

**Issue by sp2ong** [i-p-tel/sipdroid#1021]
Is it possible add opus codec ?  https://opus-codec.org/
we can use opus codec in asterisk with PC softphone like PhonerLite it will be nice to have opus codec on android sip pone

Regards

**Comment by niutech** [comment]
There is [sipdroid-pro](https://github.com/jln-ho/sipdroid-pro), which adds Opus & AAC.

**Comment by sjlongland** [comment]
@niutech Unfortunately, AAC makes it problematic.

> …a patent license is required for all manufacturers or developers of AAC codecs. For this reason, free and open source software implementations such as FFmpeg and FAAC may be distributed in source form only, in order to avoid patent infringement. (See below under Products that support AAC, Software.)

https://en.wikipedia.org/wiki/Advanced_Audio_Coding#Licensing_and_patents

`sipdroid-pro` is also not available on F-Droid, which is where I got `sipdroid` from.  Possibly for the reason mentioned above.

**Comment by jln-ho** [comment]
Original author of `sipdroid-pro` here. You can fork `sipdroid-pro` and simply remove all the AAC related stuff, should be fairly straightforward.  On second thought, it may actually make more sense to fork `sipdroid` and then port Opus support over from `sipdroid-pro`.

**Comment by sjlongland** [comment]
@jln-ho Yeah, what you've quoted there is a _software_ license, not a _patent_ license.  I avoid AMR for the same reason even though it'd guarantee better compatibility with mobiles.  Thankfully somewhere in the VoIP service chain, G.722 gets transcoded to AMR-WB anyway, so it winds up not mattering much -- the patent license becomes "not my problem".

I use AAC in other things, notably DAB+, where I do not get a choice in the matter (all the Brisbane radio stations are broadcasting using HE AAC… Want to listen?  You need to decode HE AAC), but the vast majority of the calls I receive are G.711a.  Long term though, I see this changing as more people leave the PSTN and move to VoIP and VoLTE services, so it makes sense to support some other CODECs.

Opus seems to be one that is actually being adopted.  Some devices out there also support Speex.  Given these are free and open CODECs, I'm willing to give them my support.
